### PR TITLE
Disconnect from a GitHub repository

### PR DESCRIPTION
**Pull request by MicroFish91**
Part 4 of https://github.com/microsoft/vscode-azurecontainerapps/issues/313#issuecomment-1518418209

### CODE REVIEW

File: src/commands/disconnectRepo/IDisconnectRepoContext.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ExecuteActivityContext, ISubscriptionActionContext } from "@microsoft/vscode-azext-utils";
+import { AzureSubscription } from "@microsoft/vscode-azureresources-api";
+import { IGitHubContext } from "../../gitHub/IGitHubContext";
+import { ContainerAppModel } from "../../tree/ContainerAppItem";
+
+export interface IDisconnectRepoContext extends ISubscriptionActionContext, IGitHubContext, ExecuteActivityContext {
+    targetContainer: ContainerAppModel;
```
Comment (MicroFish91) [inline src/commands/disconnectRepo/IDisconnectRepoContext.ts:12]: This pair of properties gets repeatedly assigned throughout the codebase... probably a good indicator that we need to create a uniform base context to extend from..

**Review by bwateratmsft — APPROVED** [review]

File: src/commands/disconnectRepo/IDisconnectRepoContext.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    subscription: AzureSubscription;
```
Comment (nturinski) [inline src/commands/disconnectRepo/IDisconnectRepoContext.ts:13]: It's really weird that `ISubscriptionActionContext` doesn't contain an `AzureSubscription`...

File: src/commands/registerCommands.ts [hunk 3]
```diff
@@ -41,6 +42,7 @@ export function registerCommands(): void {
 
     // github
     registerCommandWithTreeNodeUnwrapping('containerApps.connectToGitHub', connectToGitHub);
+    registerCommandWithTreeNodeUnwrapping('containerApps.disconnectRepo', disconnectRepo);
```
Comment (nturinski) [inline src/commands/registerCommands.ts:45]: `containerApps.disconnectGitHub`? Again, kind of weird to me to not use `connectToRepo` but use `disconnectRepo`.

**Review by nturinski — APPROVED** [review]

**Review by nturinski — APPROVED** [review]

File: src/commands/disconnectRepo/IDisconnectRepoContext.ts (hunk 2 above)
Comment (MicroFish91) [reply src/commands/disconnectRepo/IDisconnectRepoContext.ts:13]: Yeah we need to revisit this pattern in general so we don't need to keep adding double subscriptions

Comment (MicroFish91) [reply src/commands/disconnectRepo/IDisconnectRepoContext.ts:13]: Double...

https://www.youtube.com/watch?v=OQSNhk5ICTI

File: package.nls.json [hunk 4]
```diff
@@ -28,6 +28,7 @@
     "containerApps.editScalingRange": "Edit Scaling Range...",
     "containerApps.addScaleRule": "Add Scale Rule...",
     "containerApps.connectToGitHub": "Connect to GitHub Repository...",
+    "containerApps.disconnectRepo": "Disconnect from Repo...",
```
[the remark this answers is not in the document]
Comment (MicroFish91) [reply package.nls.json:31]: I agree with your comment on the ellipses, I should have made the change here, but I'll bundle it with the next PR, they're still somewhat related anyway :)

**Pull request merged**